### PR TITLE
Fetch optimizations for stage widgets

### PR DESCRIPTION
**Pull request by kubama**

### CODE REVIEW

File: widgets/deploymentButton/src/DeployModal.js [hunk 1]
```diff
@@ -232,13 +219,13 @@ export default class DeployModal extends React.Component {
...
                         >
-                            <Form.Dropdown
-                                search
-                                selection
+                            <DynamicDropdown
                                 value={this.state.blueprint.id}
                                 name="blueprintName"
-                                options={options}
-                                onChange={this._selectBlueprint.bind(this)}
+                                fetchUrl="/blueprints?_include=id"
```
Comment (qooban) [inline widgets/deploymentButton/src/DeployModal.js]: We don't want to sort blueprints by id similarly to sites?

File: widgets/common/src/UpdateDeploymentModal.js [hunk 2]
```diff
@@ -238,14 +223,13 @@ class UpdateDeploymentModal extends React.Component {
...
-                            <Form.Dropdown
-                                search
-                                selection
+                            <DynamicDropdown
                                 value={this.state.blueprint.id}
                                 placeholder="Select Blueprint"
                                 name="blueprintName"
-                                options={blueprintsOptions}
-                                onChange={this._selectBlueprint.bind(this)}
+                                fetchUrl="/blueprints?_include=id"
```
Comment (qooban) [inline widgets/common/src/UpdateDeploymentModal.js]: We don't want to sort blueprints by id?

File: app/app.js [hunk 3]
```diff
@@ -57,6 +57,7 @@ const browserHistory = createBrowserHistory({
 export default class app {
     static load() {
         window.React = React;
+        window.ReactDOM = ReactDOM;
```
[the remark this answers is not in the document]
Comment (kubama) [reply app/app.js]: Required by react-visibility-sensor

File: widgets/blueprintInfo/src/widget.js [hunk 4]
```diff
@@ -55,16 +55,15 @@ Stage.defineWidget({
...
-                    });
-                });
+                return Promise.all([
+                    actions.doGetBlueprintDetails(blueprintId),
+                    actions.doGetBlueprintDeployments(blueprintId)
+                ]).then(data => ({
+                    ...data[0],
+                    created_at: Stage.Utils.Time.formatTimestamp(data[0].created_at),
+                    updated_at: Stage.Utils.Time.formatTimestamp(data[0].updated_at),
+                    deployments: data[1].items[0].deployments
```
[the remark this answers is not in the document]
Comment (kubama) [reply widgets/blueprintInfo/src/widget.js]: As I understand it's not part of official language specification - for now it remains just a proposal. If so I'd rather wait until it's released before using it in production code.

File: widgets/deploymentButton/src/DeployModal.js (hunk 1 above)
Comment (kubama) [reply widgets/deploymentButton/src/DeployModal.js]: We do, it's being handled inside DynamicDropdown component. I removed explicit (and unnecessary) `_sort` parameter from sites URL so it's less confusing.

File: widgets/common/src/UpdateDeploymentModal.js (hunk 2 above)
Comment (kubama) [reply widgets/common/src/UpdateDeploymentModal.js]: See my previous comment

**Pull request merged**